gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/editor/GravatarQuickEditorPage.kt

```diff
@@ -1,43 +1,69 @@
 package com.gravatar.quickeditor.ui.editor
 
+import androidx.compose.animation.EnterTransition
+import androidx.compose.animation.ExitTransition
 import androidx.compose.foundation.layout.Box
 import androidx.compose.foundation.layout.fillMaxSize
 import androidx.compose.material3.Surface
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
-import androidx.compose.runtime.mutableStateOf
-import androidx.compose.runtime.saveable.rememberSaveable
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.tooling.preview.Preview
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
+import com.gravatar.quickeditor.ui.GravatarQuickEditorSplashPage
 import com.gravatar.quickeditor.ui.oauth.OAuthPage
 import com.gravatar.quickeditor.ui.oauth.OAuthParams
+import com.gravatar.ui.GravatarTheme
 
 @Composable
 internal fun GravatarQuickEditorPage(
     gravatarQuickEditorParams: GravatarQuickEditorParams,
     onAuthError: () -> Unit = {},
 ) {
-    val isAuthorized = rememberSaveable { mutableStateOf(false) }
+    val navController = rememberNavController()
 
-    Surface {
-        if (!isAuthorized.value) {
+    NavHost(
+        navController,
+        startDestination = "entry",
```

Comment:
> There needs to be a startDestination defined so we need something and I don't want to start with the oauth route and close it quickly if we detect that the token is present.

Why not keep the `isAuthorized` here and use it to define the `starDestination`? Something like:

```suggestion
    val isAuthorized by rememberSaveable { mutableStateOf(false) }
    NavHost(
        navController,
        startDestination = if (isAuthorized) "quickeditor" else "oauth",
```

Reply:
This will be an IO operation later so not sure we want to block the thread here 🤔